distribution/test/itests/test-itests-catalog/src/test/java/ddf/catalog/test/AbstractIntegrationTest.java

```diff
@@ -81,38 +88,114 @@
+    protected static final String CSW_SOURCE_ID = "cswSource";
 
-    protected static final String SSH_PORT = "9101";
+    protected static ServerSocket placeHolderSocket;
 
-    protected static final String RMI_SERVER_PORT = "44445";
+    protected static Integer basePort;
 
-    protected static final String RMI_REG_PORT = "1100";
+    protected static final String DDF_HOME_PROPERTY = "ddf.home";
 
-    protected static final String SECURE_ROOT = "https://localhost:";
+    protected static String ddfHome;
 
-    protected static final String SERVICE_ROOT = SECURE_ROOT + HTTPS_PORT + "/services";
+    public enum DynamicPort {
+        BASE_PORT("org.codice.ddf.system.basePort"), HTTP_PORT(
+                "org.codice.ddf.system.httpPort"), HTTPS_PORT(
+                "org.codice.ddf.system.httpsPort"), SSH_PORT, RMI_SERVER_PORT, RMI_REG_PORT;
 
-    protected static final String INSECURE_ROOT = "http://localhost:";
+        private final String systemProperty;
 
-    protected static final String INSECURE_SERVICE_ROOT = INSECURE_ROOT + HTTP_PORT + "/services";
+        DynamicPort() {
+            this.systemProperty = null;
+        }
 
-    protected static final String REST_PATH = SERVICE_ROOT + "/catalog/";
+        DynamicPort(String systemProperty) {
+            this.systemProperty = systemProperty;
+        }
 
-    protected static final String OPENSEARCH_PATH = REST_PATH + "query";
+        String getSystemProperty() {
+            return this.systemProperty;
+        }
 
-    protected static final String CSW_PATH = SERVICE_ROOT + "/csw";
+        public String getPort(Integer basePort) {
+            return String.valueOf(basePort + this.ordinal());
+        }
 
-    protected static final String OPENSEARCH_SOURCE_ID = "openSearchSource";
+        public String getPort() {
+            return String.valueOf(basePort + this.ordinal());
+        }
+    }
 
-    protected static final String CSW_SOURCE_ID = "cswSource";
+    public static class DynamicUrl {
+        public static final String SECURE_ROOT = "https://localhost:";
 
-    protected static final String DDF_HOME_PROPERTY = "ddf.home";
+        public static final String INSECURE_ROOT = "http://localhost:";
 
-    protected static String ddfHome;
+        private final String root;
+
+        private final String tail;
+
+        private final DynamicPort port;
+
+        private final DynamicUrl url;
+
+        public DynamicUrl(String root, DynamicPort port) {
+            this(root, port, "");
+        }
+
+        public DynamicUrl(String root, DynamicPort port, String tail) {
+            this.root = root;
+            this.port = port;
+            this.url = null;
+            this.tail = tail;
+        }
+
+        public DynamicUrl(DynamicUrl url, String tail) {
+            this.root = null;
+            this.port = null;
+            this.url = url;
+            this.tail = tail;
+        }
+
+        public String getUrl() {
+            if (null != port) {
```

Comment:
Couldn't someone (incorrectly) invoke one of the `root`, `port` flavored ctors with a null port? And wouldn't that then mess up the logic here?

Maybe `@NotNull` annotations or `IllegalArgumentException`s?